jetbrains-core/src/software/aws/toolkits/jetbrains/services/lambda/upload/CreateFunctionDialog.kt

```diff
@@ -70,13 +73,21 @@ class CreateFunctionDialog(private val project: Project, private val initialRunt
     override fun doValidate(): ValidationInfo? = view.validatePanel()
 
     override fun doCancelAction() {
-        LambdaTelemetry.deploy(
-            project,
-            result = Result.Cancelled,
-            regionId = project.activeRegion().id,
-            lambdaPackageType = LambdaPackageType.from(view.configSettings.packageType().toString()),
-            initialDeploy = true
-        )
+        val connectionSettings = AwsConnectionManager.getInstance(project).connectionSettings()
+        val account = connectionSettings?.let { AwsResourceCache.getInstance().getResourceIfPresent(StsResources.ACCOUNT, it) }
+        account?.let { it ->
+            MetricEventMetadata(
+                awsAccount = it,
+                awsRegion = connectionSettings.region.toString()
+            )
+        }?.let { metricEventMetadata ->
```

Comment:
We should always push the telemetry, even if we can't get the info.

See:
```
software.aws.toolkits.jetbrains.services.telemetry.TelemetryService#record
```

Reply:
Maybe we should expose this logic as a public method on TelemetryService to encapsulate it,
something like

`TelemetryService#calcualteMetadata(project: Project, connectionSettingsOverride: ConnectionSettings? = null) `

Reply:
Re-reading, why are we making the metadata manually when this is just the project settings?

Reply:
Changed it to project, wasn't sure if the region was accounted for earlier